via/blocker.py

```diff
@@ -46,35 +48,159 @@ def __init__(self, application, checkmate_host=None):
         self._checkmate = CheckmateClient(checkmate_host)
 
     def __call__(self, environ, start_response):
-        url_to_annotate = wsgi.get_path_info(environ)[1:]
+        # Parse the URL and referrer
+        request_path = wsgi.get_path_info(environ)
+        via_host = environ.get("HTTP_HOST")
+        classified_url = ClassifiedURL.classify(request_path, via_host, assume_via=True)
+        classified_referrer = ClassifiedURL.classify(
+            environ.get("HTTP_REFERER"), via_host
+        )
+
+        # Determine which urls should be run against checkmate (if any)
+        full_check, partial_check, rule_type = self._get_urls_to_check(
+            classified_url, classified_referrer
+        )
+
+        # Apply the checks
+        for url_to_check, allow_all in ((full_check, False), (partial_check, True)):
+            if not url_to_check:
+                # No check of this type requested
+                continue
+
+            hits = self._check_url(url_to_check.proxied_url, allow_all=allow_all)
+            if not hits:
+                continue
+
+            response = self._render_block_template(hits, url_to_check)
+            return response(environ, start_response)
+
+        return self._application(environ, start_response)
+
+    @classmethod
+    def _get_urls_to_check(cls, url, referrer):
+        """Determine what checks to make based on a URL and referrer.
 
-        if not url_to_annotate:
-            # This is our own landing page "/", no need to check this
-            return self._application(environ, start_response)
+        This will return up to two of the passed in ClassifiedURL objects. The
+        first of which should have a full allow list enabled check. The second
+        should get a partial allow list disabled check.
```

Comment:
```suggestion
        Return a tuple of three items:
 
        1. A URL that should be "fully checked" (checked against both Checkmate's
           allow-list and its block-list).
           This will be one of the given `url` or `referrer`, or `None`
        2. A URL that should be "partially checked" (checked against Checkmate's
           block-list only).
           This will also be one of the given `url` or `referrer`, or `None`
        3. The name of the triggered rule (for debugging)
```

Making this a bit more explicit (particularly about what "fully checked" and "partially checked" mean, defining the terms)